### DISCUSSION TITLE
Duplicate component schemes workaround. But the new duplicates simply reference ($ref) to already existing ones

### DISCUSSION THREAD

**Discussion by ValdemarBirkegaardPetersen** [microsoft/OpenAPI.NET#846] [General]
'm trying to merge several OpenAPI specs into a single OpenAPI document. However, some of the OpenAPI specs that im trying to merge, contains Components.Schemes with identical key names but different values/schemes. My initial solution was to check if a key already exists, and if it does, i simply change the key name to "KeyName2","KeyName3", and so on. This is the current code:

```
  // Components --> Schemas
  int iterator = 2
  foreach (var schemas_ite in importedOpenApiDocument.Components.Schemas)
  {
        if (document.Components.Schemas.ContainsKey(schemas_ite.Key))
          {
		       string newSchemesString = schemas_ite.Key + iterator;
        	   document.Components.Schemas.Add(newSchemesString, schemas_ite.Value);
               iterator += 1
          }
          else
          {
               document.Components.Schemas.Add(schemas_ite.Key, schemas_ite.Value); 
          }
  }
```
        
When im serializing the merged OpenAPI spec, the new keys exists as expected, but the values of the new keys are simply references to the previous identical key, even though the value/content of the scheme is different. 

Output from the merged OpenAPI document is below:

```
components:
  schemas:
    ProblemDetails:
      type: object
      properties:
        title2:
    Subscription:
      type: string
    ProblemDetails2:
      $ref: '#/components/schemas/ProblemDetails'
    Subscription2:
      $ref: '#/components/schemas/Subscription'
  
```

As you can see, the ProblemDetails2 and Subscription2 is created with a reference ($ref)  to the schemes that alreay have been added to the final document. Even though some of the schemes have identical names, their content and structure is very different.  How do i fix this issue? I know there is function called SerializeAsV3WithoutReferences, but thats really my only idea. Can anyone point me in the right direction?

**Comment by ValdemarBirkegaardPetersen** [comment]
Here is an image of the OpenAPI .yml files im testing this with. Here you can clearly see that the content of the duplicate schemes are different. I want them to created as "NewScheme2", "NewScheme3" with unique values inside, not just references to already existing ones, which are incorrect. 
![ForGITHUB](https://user-images.githubusercontent.com/56254921/166166781-89606fd3-7f1b-4d77-bf3e-019494085a02.png)

**Comment by baywet** [comment]
Hi @ValdemarBirkegaardPetersen

Thank you for using the SDK and for reaching out.

Your merging code would need to handle the references updates. As it's a more complete solution I don't have a code example to share.

But essentially the steps would be:
1. run name collision detection for both documents
2. rename the colliding schema components in place in the source document
3. update the references in place in the source document
4. create a merged document from both sources

> Note: this is not limited to just the schemas, any component kind need to go through that logic.

Let us know if you have any additional comments or questions.
